test/e2e_new/aws_test.go

```diff
@@ -0,0 +1,80 @@
+// +build e2e
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_new
+
+import (
+	"context"
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
+	"path/filepath"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	"sigs.k8s.io/cluster-api/util"
+)
+
+var _ = Describe("functional tests", func() {
```

Comment:
@gab-satchi if the goad of this test is simply to spin un a workload cluster, you can consider re-using the test spec from [test/e2e](https://github.com/kubernetes-sigs/cluster-api/blob/master/test/e2e/quick_start.go). You can find an example of how the entire test spec can be re-used [here](https://github.com/kubernetes-sigs/cluster-api/blob/c3ef9222bb61524f5d3705ca1bf1202f3afeb720/test/e2e/quick_start_test.go#L27-L39).

Also other specs are available: self-hosting, upgrades, machine health checks...

Reply:
The thought of reusing quickstart crossed my mind for this along with another test in CAPI but I dismissed it because I couldn't configure any of the cluster parameters such as name in this case. One way to configure these parameters would be for e2e_config to expose a setter to change variables once the configuration has been loaded. This will make it easier for multiple similar but slightly different 'quick-start' tests to reuse the spec without breaking the simplicity of the quick-start spec. WDYT?

Reply:
Name is a poor example for this case but ability to modify other template variables could be useful

Reply:
@gab-satchi if you can point out what you want to configure, I will be happy to follow up.
WRT to template variables, you can drive them via config file or via is variables

Reply:
For the AWS tests in question, name is the only one that needs to be configurable during cluster creation. The rest are all about modifying resources afterwards which can be done with helpers. I think I'm missing something conceptually here WRT reusing the existing capi specs. 

My understanding of how CAPA can use the framework for a test like "testing load balancer service type": 
- Use framework to setup the workload cluster
- Create a service of type LoadBalancer (can be done with update to helpers)
- Use framework to delete workload cluster
- Assert on infra resources (LB) getting deleted

My understanding with using the existing capi spec: quick-start will create a cluster, wait for it to be created then delete the cluster immediately after. I'm not sure how CAPA can interject in the middle to make any assertions. Leaving the cleanup of the cluster to the caller is an option.

Reply:
Resolved offline. CAPI e2e specs with AWS can be used as a separate prow job if necessary. Changes to CAPA e2e can use helpers in the framework and not the full specs as CAPA would need some intermediate AWS calls.